docs/dns-usage.rst

```diff
@@ -0,0 +1,184 @@
+Using the API
+=============
+
+Authorization / Configuration
+-----------------------------
+
+- Use :class:`Client <gcloud.dns.client.Client>` objects to configure
+  your applications.
+
+- :class:`Client <gcloud.dns.client.Client>` objects hold both a ``project``
+  and an authenticated connection to the PubSub service.
+
+- The authentication credentials can be implicitly determined from the
+  environment or directly via
+  :meth:`from_service_account_json <gcloud.dns.client.Client.from_service_account_json>`
+  and
+  :meth:`from_service_account_p12 <gcloud.dns.client.Client.from_service_account_p12>`.
+
+- After setting ``GOOGLE_APPLICATION_CREDENTIALS`` and ``GCLOUD_PROJECT``
+  environment variables, create an instance of
+  :class:`Client <gcloud.dns.client.Client>`.
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client()
+
+- Override the credentials inferred from the environment by passing explicit
+  ``credentials`` to one of the alternative ``classmethod`` factories,
+  `:meth:gcloud.dns.client.Client.from_service_account_json`:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client.from_service_account_json('/path/to/creds.json')
+
+  or `:meth:gcloud.dns.client.Client.from_service_account_p12`:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client.from_service_account_p12('/path/to/creds.p12', 'jrandom@example.com')
+
+
+Projects
+--------
+
+A project is the top-level container in the ``DNS`` API:  it is tied
+closely to billing, and can provide default access control across all its
+datasets.  If no ``project`` is passed to the client container, the library
+attempts to infer a project using the environment (including explicit
+environment variables, GAE, and GCE).
```

Comment:
Really more `environment variables, GAE, or GCE` than `environment variables, GAE, and GCE`

But it's a nit, so you can leave it if you like.